### PR TITLE
Add wake_word integration docs

### PR DESCRIPTION
**Pull request by synesthesiam**
## Proposed change

Document new `wake_word` integration: https://github.com/home-assistant/core/pull/96380
- can be merged once text fragment is merged: #29129

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by c0ffeeca7 — DISMISSED** [review]
LGTM, @synesthesiam :+1:

**Comment by synesthesiam** [comment]
Please wait to merge until https://github.com/home-assistant/core/pull/96380 has been merged

**Comment by synesthesiam** [comment]
Ready to merge

**Comment by frenck** [comment]
Marking this one as a draft, as we are not going to actively publish this for 2023.9

File: source/_integrations/wake_word.markdown [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ha_domain: wake_word
+ha_integration_type: entity
+---
+
+A wake-word-detection entity allows other integrations or applications to detect wake words in streaming audio.
+
+The wake-word-detection entities cannot be implemented manually, but can be provided by integrations such as [Wyoming](/integrations/wyoming). The API in [Assist Pipelines](https://developers.home-assistant.io/docs/voice/pipelines/) enables wake-word-detection as part of [Assist](/voice_control/).
+
+
+{% include integrations/config_flow.md %}
```
Comment (frenck) [inline source/_integrations/wake_word.markdown]: This cannot be set up like that for entity components?

Comment (c0ffeeca7) [inline source/_integrations/wake_word.markdown]:
```suggestion

{% include integrations/building_block_integration.md %}
```

**Review by frenck — APPROVED** [review]

**Pull request merged**